### PR TITLE
2.6 doc restructure

### PR DESCRIPTION
**Pull request by johanandren**
## Purpose
Restructure the table of contents and make the Akka Typed APIs the default.

## References
Refs #26487 

## Changes
* Make the typed APIs the default
* Move the old Actor APIs to a "Classic" subtree
* Clearly mark up old docs and link to the corresponding new pages

## Background Context
For 2.6 we want the default APIs for new (and old) users to be the new typed APIs, we also want to keep the old Actor API docs for existing applications.

### CODE REVIEW

File: akka-docs/src/main/paradox/actors.md [hunk 1]
```diff
@@ -1,8 +1,11 @@
-# Actors
+# Classic Actors
+
+Akka Classic is the original Actor APIs now superseeded by the new more type safe actor APIs.
```
Comment (patriknw) [inline akka-docs/src/main/paradox/actors.md]: maybe not say superseded since it has a negative connotation 

Suggestion:

Akka Classic is the original Actor APIs, which have been improved by more type safe and guided Actor APIs, known as Akka Typed. Akka Classic is still fully supported and existing applications can continue to use the classic APIs. It is also possible to use Akka Typed together with classic actors within the same `ActorSystem`, see <link to coexisting.md>. For new projects we recommend using the new Actor APIs.

btw, also typo in superseeded

File: akka-docs/src/main/paradox/fsm.md [hunk 2]
```diff
@@ -1,4 +1,7 @@
-# FSM
+# Classic FSM
+
+Akka Classic is the original Actor APIs now superseeded by the new more type safe actor APIs.
+For the new API see @ref[fsm](typed/fsm.md).
 
 ## Dependency
```
Comment (patriknw) [inline akka-docs/src/main/paradox/fsm.md]: redirect for removed getting started, I think it should redirect to the typed guide

**Review by patriknw — COMMENTED** [review]
looking good, I like this structure

in follow up PR we will also try to reduce number of top level topics, e.g. by collapsing Additional information and Project information, and remove How to.

File: akka-docs/src/main/paradox/actors.md (hunk 1 above)
Comment (johanandren) [reply akka-docs/src/main/paradox/actors.md]: Sounds good, I'll use that.

**Comment by TimMoore** [comment]
Looks good, happy to see this! 👍 

Here are a few things I noticed while browsing through (maybe out of scope for this pull request). Sorry for the brain dump, but I thought it would be better to get this all down.

- The "General Concepts" still assumes classic:
  - The "Actor Systems" page might want to mention the explicit single root actor required in Akka Typed
  - "What is an Actor?" talks about using `become` and `unbecome` to change behaviors, and `context.actorOf` to create children
  - I'm not sure whether the top-level hierarchy described in "Supervision and Monitoring" is still accurate for typed systems, and other details need updates
  - So on in the rest of the pages… I won't keep listing everything
- It's good that the classic docs now reference the typed ones, but we might want to acknowledge the classic API in the typed docs as well, to make sure people are pointed to the right place for their needs.
  - For example, the typed actors page starts with: "To use Akka Actor Typed, you must add the following dependency in your project". This won't mean much to anyone who doesn't know the historical context.
  - Anyone new to Akka but working on an existing system using classic APIs (e.g., starting a new job with a company using Akka) might not easily find the right place in the docs to understand the existing code they see.
  - In particular, the "Relation to Akka (untyped) Actors" section probably needs a little rework, and the "Coexistence" introduction should provide more context (maybe even move to an appendix or to the "classic" section).
- The "Dispatchers" section still says "For full details on how to configure custom dispatchers see the untyped docs." Maybe those should be swapped.
- "Actor discovery" starts by referencing the way you do it in untyped actors, which is only useful when that's something you already understand. Still worth noting, maybe just not as the first thing.
- Similarly, the "Streams" page _under_ the (typed) "Actors" section is positioned as an alternative rather than the default. The whole page should be integrated into the main "Streams" section.
- "Behaviors as Finite state machines", "Persistence", "Cluster", "Cluster Sharding" and others also start by referencing untyped.
- "Persistence" comes before "Akka Cluster" (and therefore "Cluster Sharding") in the ToC, but references how it's used with Cluster Sharding. Maybe better to present in the reverse order?
- "Cluster Specification" is still under the classic section, but should probably be moved to typed
- Maybe rather than having "Classic" in a separate subtree, it would be better to have "Classic" pages within the "Actors", "Akka Cluster", "Persistence", etc. sections.
- It would be good to eventually update all referenced code snippets to typed (e.g., https://github.com/akka/akka/blob/master/akka-discovery/src/test/scala/doc/akka/discovery/CompileOnlySpec.scala) so that people who click through to the source will see the right packages to import.
- Have we considered taking "typed" out of the URL and having the typed pages replace the existing URLs (with redirects from current typed URLs)?
  - That way, if you were to land on (say) https://doc.akka.io/docs/akka/2.5.21/actors.html at some point after Akka 2.6 is released, the link to the latest version in the warning at the top will take you to the typed page rather than the classic one.
  - This will probably also help with canonical references to make sure the typed pages get boosted in search results (but I guess we would also need to special case it for existing typed documentation so that the canonical URLs don't redirect).
  - As a tangent, the same idea could apply for Artery vs. Classic remoting

Summing up, there are three different audiences to consider:

1. People brand new to Akka, starting a new system, who we want to present with Typed as the default way to learn
2. People brand new to Akka, trying to understand an existing system (likely using Classic) who we want to be able to find the relevant documentation easily
3. People familiar with Classic, trying to learn Typed, who want to see succinctly what's changed without rehashing the basic concepts

All three of these audiences are important, but it might be hard to serve them all with one set of pages. The main thing is to try to quickly route everyone to the right place, no matter where they first land in the docs (e.g., from a Google search or link from a blog post).

**Comment by johanandren** [comment]
Thanks for the input Tim it is useful. 

I think we'll try to complete this work over several PRs and this will be mostly about changing the structure.

**Comment by patriknw** [comment]
Thanks for detailed comments. We will follow up with content changes in separate issues and PRs.

**Review by chbatey — APPROVED** [review]
LGTM, I had a navigate around and I like the structure. Might want to update all the "classic" pages to remove superseded

**Review by patriknw — APPROVED** [review]
LGTM

**Comment by raboof** [comment]
`Error writing content for page scheduler.html: Unknown anchor [typed/interaction-patterns.html#typed-scheduling] referenced from [scheduler.html]` - finally what looks like a bonafide warning :laughing: 

Did you mean to also commit a `akka-classic-header.md`?

**Comment by johanandren** [comment]
Heh, no I meant to revert that change and just put the header everywhere. But my search-replace-skills must have failed me.

**Comment by patriknw** [comment]
@TimMoore I have incorporated your suggestions in the task list here https://github.com/akka/akka/issues/24717#issuecomment-510842580 and here https://github.com/akka/akka/issues/27223#issue-461040548

There are two things we disagree with.

> Maybe rather than having "Classic" in a separate subtree, it would be better to have "Classic" pages within the "Actors", "Akka Cluster", "Persistence", etc. sections.

We considered that when discussing the structure but dismissed it because
* there will be too many sub-sections in the left ToC
* if you are navigating the ToC you already know if you are looking for Typed or Classic and shouldn't have to make that choice more than once
* by having Typed docs as the main ToC we promote that to be the recommended for new projects, and existing users know where to find all of the Classic specific stuff

> Have we considered taking "typed" out of the URL and having the typed pages replace the existing URLs (with redirects from current typed URLs)?

We thought that would be to disruptive to all existing links. Many of those link to a very specific heading in the page (and many of the Classic pages are very long, so difficult for users to find the right section if they ended up in the Typed docs and it doesn't have the same heading)

Google SEO rules when it comes to redirects and canonicals seems to be some black magic so it's not sure it will work as we would prefer.

I think we should start without that and we can revisit if it turns out to be a problem that users land on wrong pages (and we want to push Typed harder as the primary choice).
